keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -0,0 +1,1021 @@
+each cluster, such as the number of nodes, their roles, and their
+current status. This can help administrators manage the clusters more
+effectively and ensure that they are always operating at optimal levels.
+
+#### Story 3: Transparent to Consumers
+
+The unified API ensures that consumers have the flexibility to choose
+different cluster management tools and switch among them as needed.
+Regardless of the tool they choose, they all use the same API to define
+clusters.
+
+This means that consumers can switch from one cluster manager to another without
+having to worry about compatibility issues or learning a new API. This
+can significantly reduce the learning curve and make it easier for
+consumers to adopt new tools.
+
+Moreover, the unified API can also provide a consistent user experience
+across different cluster managers. For example, if a consumer is used to a
+particular command or function in one tool, they can expect the same
+command or function to work in the same way in another tool. This can
+further enhance the usability and adoption of different cluster
+manager.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+We try to summarize the most common properties that should be able to
+represent a cluster and support the above use case at the minimum scope.
+
+The target consumers of the API are users who manage the clusters and
+other tools to understand the clusters concept from the API for
+multicluster scheduling, workload distribution and cluster management.
+The API aims to provide the information for the consumers to answer the
+below questions:
+
+* Is the cluster under management?
+* How can I select a cluster?
+* Is the cluster healthy?
+* Does the cluster have certain capabilities or properties?
+* Does the cluster have sufficient resources?
+
+### Cluster Name
+
+It is required that cluster name is unique for each cluster, and it
+should also be unique among different providers (cluster manager). There
+are two option to ensure it is uniqueness
```

Comment:
So long as clusters reside in the same namespace (or aren't namespaced), name uniqueness is already ensured. Why do we need to force certain patterns? I assume every cluster manager already has some naming system, and it seems like it would be good to use whatever is already provided?

Reply:
This has been a topic of debate on every call about this (Sig-MC meeting). I feel we should drop this from the initial v1alpha1 and see if the community needed this at the end for the next iteration of the API

Reply:
changed the wording here. It is provides the responsibility to ensure the name uniqueness. The proposal only provides two possible options but not mandatory.